### DISCUSSION TITLE
SKR MINI E3 V3.0

### DISCUSSION THREAD

**Discussion by IsNotMyUserName** [grblHAL/STM32F4xx#160] [General]
Hello!
I have the SKR MINI E3 V3.0 and I would like to know if you plan to add support for this board?

It has a different stm32...

**Comment by terjeio** [comment]
I have no plans to add it but may do so if you can create a [board map](https://github.com/grblHAL/STM32F1xx/blob/master/Inc/btt_skr_mini_e3_2.0_map.h) for it.

**Reply by IsNotMyUserName** [reply]
Ok, I'll see what I can do.

**Comment by djwmarcx** [comment]
Hi! I picked this up and ported grblHAL to the STM32G0: https://github.com/djwmarcx/STM32G0xx

It targets the BTT SKR Mini E3 V3.0 (STM32G0B1RET6) and is verified on real hardware — my CNC router runs on it: stepper motion, TMC2209 UART (all four drivers via USART4), native USB CDC, hard limits, probe, spindle PWM, flash-based settings storage, and a ganged dual-motor Y axis. The stock BTT SD-card bootloader is preserved (app linked at 0x08002000), so it can also be flashed the usual SD-card way.

It's adapted from the STM32F1xx driver, so the structure should look familiar. Porting notes are in the README (no bit-banding on the M0+, the G0's split EXTI pending registers and shared vectors, USB_DRD_FS clocked from HSI48+CRS, and the half-duplex TMC UART quirk from STM32F1xx#49, which applies here too).

The core only needs a 5-line change in platform.h to add STM32_G0_PLATFORM detection — I'll open a PR in the future (once stable) for that from this branch (https://github.com/djwmarcx/core/tree/stm32g0-platform). SD card, I2C/EEPROM and keypad are not ported yet.

Still early days (one board, one machine), but it works and I'd be happy for others to test it — and for it to be listed as a third-party driver if you think it fits, or adopted into the org if there's interest.

**Comment by IsNotMyUserName** [comment]
Nice, I never got around tinker with it. I have a mini board and the display for it as well. I can give it a go later.
